### ISSUE TITLE
Select field type should accept an array of objects

### ISSUE THREAD

**Issue by sjstark** [BKWLD/vue-form#3]

**Comment by weotch** [comment]
That certainly makes sense to me ... it looks like it's expecting an array of strings right now?  @jonjahr what does this regex do?

https://github.com/BKWLD/vue-form/blob/30e7a42eb7194b075b8a4365507a9ccd16e50b76/src/components/select-field.vue#L76

Like is the expected string value of an option currently?  

Maybe, to preserve backwards compatibility, we can `typeof` check the value of an option and if string do the regex thing else if object, expect one with a structure like @sjstark's suggestion.

**Comment by jonjahr** [comment]
Yes this makes sense to me too. The regex I’m pretty sure tries to parse a label-value pair from the string with a pipe separator, ie “Layout Left | layout-left”.  I’d like to keep the string option for back compatibility at least for a little while until I know for sure the POJO route is easy to work with in Contentful.

**Comment by weotch** [comment]
Ahhh, it's like that for Contentful, gotcha.

Btw, did you guys just learn about `POJO` recently?  Cause I did. 🤓
